loader/src/sources/vts/geo.js

```diff
@@ -0,0 +1,146 @@
+const got = require("got");
+const Sentry = require("@sentry/node");
+
+const dataProviders = {
+  "Rite-Aid": {
+    provider: "rite_aid",
+    getStoreName: (data) => {
+      const m = data.name.match(/RITE AID PHARMACY #?(\d+)/i);
+      if (m) {
+        return `Rite Aid #${parseInt(m[1], 10)}`; // format as in riteaid/api.js
+      }
+      return data.name;
+    },
+  },
+  CVS: {
+    // modeled after code in cvs/api.js
+    provider: "cvs",
+    getStoreName: (data) => {
+      const m = data.name.match(/CVS Pharmacy, Inc. #?(\d+)/i);
+      if (m) {
+        return `CVS #${parseInt(m[1], 10)}`; // format as in cvs/api.js
+      }
+      return data.name;
+    },
+  },
+};
+
+function warn(message, context) {
+  console.warn(`VTS Geo: ${message}`, context);
+  Sentry.captureMessage(message, Sentry.Severity.Info);
+}
+
+function error(message, context) {
+  console.error(`VTS Geo: ${message}`, context);
+  Sentry.captureMessage(message, Sentry.Severity.Error);
+}
+
+async function queryState(state) {
+  try {
+    const response = await got({
+      url: `https://api.vaccinatethestates.com/v0/${state}.geojson`,
+    });
+    return JSON.parse(response.body);
+  } catch (error) {
+    error(`Error fetching Vaccine Spotter data`, error);
+    return [];
+  }
+}
+
+/**
+ * Filter out some stores with bad data.
+ * @param {any} store
+ * @returns {boolean}
+ */
+function hasUsefulData(store) {
+  return (
+    store.properties.concordances?.length &&
+    store.geometry?.coordinates &&
+    store.properties.provider?.name &&
+    store.properties.provider?.name in dataProviders
+  );
+}
+
+function formatStore(store) {
+  const data = store.properties;
+
+  let result;
+  Sentry.withScope((scope) => {
+    const dataProvider = dataProviders[data.provider.name];
+    if (!dataProvider) {
+      error(`Unexpected provider name ${data.provider.name}`);
+      return null;
+    }
+
+    const provider = dataProvider.provider;
+    const name = dataProvider.getStoreName(data);
+
+    scope.setContext("location", {
+      id: store.id,
+      url: data.vts_url,
+      name: name,
+      provider: provider,
+    });
+
+    function splitConcordance(concordance) {
+      const colon = concordance.indexOf(":");
+      return [
+        concordance.substring(0, colon),
+        concordance.substring(colon + 1),
+      ];
+    }
+
+    const systemsToSend = new Set([
+      "cvs",
+      "rite_aid",
+      "vaccinefinder_org",
+      "vaccinespotter_org",
+    ]);
+
+    const concordances = data.concordances.map(splitConcordance);
+    const externalIds = concordances.filter((v) => systemsToSend.has(v[0]));
```

Comment:
We need a mechanism for remapping names here — in our external IDs, `vaccinespotter_org` is just `vaccinespotter` (e.g. http://getmyvax.org/api/edge/locations/vaccinespotter:141533984).

(While we’re at it, I wonder if we should rename `vaccinefinder_org` to `vaccines_gov`? Probably not a huge deal.)

Reply:
Also realized some locations use `vaccinefinder_org` while others use `vaccinefinder` as the concordance, and some even use both.